ccm_web/client/src/components/FileUpload.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { ChangeEvent, useRef } from 'react'
+import { Button, Card, CardContent, makeStyles, Typography } from '@material-ui/core'
+import CloudUploadIcon from '@material-ui/icons/CloudUpload'
+import { useSnackbar } from 'notistack'
+
+interface FileUploadProps {
+  onUploadComplete: (file: File) => void
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    textAlign: 'center'
+  },
+  cardContent: {
+    cursor: 'pointer',
+    height: 250,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center'
+  },
+  actionsBar: {
+    backgroundColor: '#EEEEEE'
+  },
+  actionsBarIcon: {
+    verticalAlign: 'middle'
+  },
+  input: {
+
+  },
+  uploadIcon: {
+    color: '#3F648E'
+  }
+}))
+
+function FileUpload (props: FileUploadProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
+  const handleFileSelected = (file: File|undefined): void => {
+    if (file === undefined) return
+    if (file.name.toUpperCase().match(/.+.CSV/) == null) {
+      enqueueSnackbar('File type not supported.', {
```

Comment:
The error message can be better, `Only supports .csv files and not xlsx`. This is another clear indication apart from stuff we provide in documentation

Reply:
If we want to  change the message I would  simply say 'Only .csv files are supported' rather than start to name file extensions that aren't :)